enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -3976,6 +3979,150 @@ class AdjointGenerator
       return;
     }
 
+    if ((funcName == "cblas_ddot" || funcName == "cblas_sdot") &&
+        called->isDeclaration()) {
+      Type *innerType;
+      std::string dfuncName;
+      if (funcName == "cblas_ddot") {
+        innerType = Type::getDoubleTy(call.getContext());
+        dfuncName = "cblas_daxpy";
+      } else if (funcName == "cblas_sdot") {
+        innerType = Type::getFloatTy(call.getContext());
+        dfuncName = "cblas_saxpy";
+      } else {
+        assert(false && "Unreachable");
+      }
+      Type *castvals[2] = {call.getArgOperand(1)->getType(),
+                           call.getArgOperand(3)->getType()};
+      auto *cachetype = StructType::get(call.getContext(), ArrayRef(castvals));
+      Value *undefinit = UndefValue::get(cachetype);
+      Value *cacheval;
+      auto in_arg = call.getCalledFunction()->arg_begin();
+      in_arg++;
+      Argument *xfuncarg = in_arg;
+      in_arg++;
+      in_arg++;
+      Argument *yfuncarg = in_arg;
+      bool xoverwrite = uncacheable_args.find(xfuncarg)->second;
+      bool yoverwrite = uncacheable_args.find(yfuncarg)->second;
+      bool xcache =
+          !gutils->isConstantValue(call.getArgOperand(3)) &&
+          (xoverwrite ||
+           cast<ConstantInt>(call.getArgOperand(2))->getSExtValue() != 1);
+      bool ycache =
+          !gutils->isConstantValue(call.getArgOperand(1)) &&
+          (yoverwrite ||
+           cast<ConstantInt>(call.getArgOperand(4))->getSExtValue() != 1);
+      if ((Mode == DerivativeMode::ReverseModeCombined ||
+           Mode == DerivativeMode::ReverseModePrimal) &&
+          (xcache || ycache)) {
+        BuilderZ.SetInsertPoint(gutils->getNewFromOriginal(&call));
+        Value *arg1, *arg2;
+        auto size = ConstantExpr::getSizeOf(innerType);
+        if (xcache) {
+          auto dmemcpy = getOrInsertDifferentialMemcpyStrided(
```

Comment:
Change this to getOrInsertMemcpyStrided (since here it shouldn't be doing the differential memcpy, just a strided memcpy).